xc/common/utils/prjxray_routing_import.py

```diff
@@ -698,39 +698,38 @@ def import_graph_nodes(conn, graph, node_mapping, connection_box_map):
             right_graph_node_pkey
         ) = result
 
-        side = node.loc.side
-        if side == tracks.Direction.LEFT:
-            assert left_graph_node_pkey is not None, (tile_type, pin_name)
-            node_mapping[left_graph_node_pkey] = node.id
+        # VPR emits only one node for each site pin, instead of one node for each
```

Comment:
This code should consult with input arch to determine which sides the node is on.  See https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1577

Reply:
This can be tracked in a follow up issue, it doesn't have to be fixed in this PR.  Currently upstream VPR is broken in this area, so it might blow up at any time :(

Reply:
You still haven't fixed this?

Reply:
Shouldn't this already be checked with the left/right/top/bottom graph node pkeys?

Reply:
That checks the wrong direction, right?  The database is based on the arch-defs side, we need something from the rr graph side to line up.  Once https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1577 is fixed, this might revert to just using something from the rr graph.  Maybe factor the check into something like "is node X on side Y", and then it can be updated once https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1577 is fixed?

Reply:
I have added a check to verify the correctness of the side for a given node, using the arch.xml as input to get the side data.

Reply:
Great.